CMakeLists.txt

```diff
@@ -11,13 +11,58 @@ if (LUA_VERSION MATCHES "5.1")
   Find_Package(Lua 5.1 EXACT REQUIRED)
 elseif(LUA_VERSION MATCHES "5.2")
   Find_Package(Lua 5.2 EXACT REQUIRED)
-elseif(NOT LUA_VERSION OR LUA_VERSION MATCHES "5.3")
+elseif(LUA_VERSION MATCHES "5.3")
   Find_Package(Lua 5.3 EXACT REQUIRED)
+else()
+  message(FATAL_ERROR "${LUA_VERSION} is not a supported version for lua.")
 endif()
 
 include_directories(${LUA_INCLUDE_DIRS})
 
-file(GLOB EXAMPLES_SRC examples/*.cpp)
+if (NOT MSVC)
+  set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wno-unknown-warning -Wno-unknown-warning-option -Wall -Wextra -Wpedantic -pedantic -pedantic-errors -Wno-noexcept-type -ftemplate-depth=1024")
```

Comment:
Is this tested for both gcc and clang? Maybe seperate release and debug flags?

Reply:
gcc and clang both need all of these warning flags: there's 2 versions of unknown warning specifically because clang and gcc don't agree with each other. XCode uses a variant of appleclang but the flags remain the same.